components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/listener/IDPMgtAuditLogger.java

```diff
@@ -67,16 +67,37 @@ public boolean doPostUpdateIdP(String oldIdPName, IdentityProvider identityProvi
         return true;
     }
 
+    /**
+     * Post delete of IDP.
+     *
+     * @param idPName Name of the IDP
+     * @param tenantDomain Tenant Domain
+     * @return
+     * @throws IdentityProviderManagementException
+     */
     @Override
     public boolean doPostDeleteIdP(String idPName, String tenantDomain) throws IdentityProviderManagementException {
-        if(StringUtils.isEmpty(idPName)){
+        if (StringUtils.isEmpty(idPName)) {
             idPName = "Undefined";
         }
         audit.info(String.format(AUDIT_MESSAGE, getUser(), "delete", UserCoreUtil.addTenantDomainToEntry
                 (idPName, tenantDomain), null, SUCCESS));
         return true;
     }
 
+    /**
+     * Additional actions after deleting IdPs of a given tenant id.
+     *
+     * @param tenantDomain Tenant domain to delete IdPs
+     * @return
+     * @throws IdentityProviderManagementException
+     */
+    @Override
+    public boolean doPostDeleteIdPs(String tenantDomain) throws IdentityProviderManagementException {
+
+        return super.doPostDeleteIdPs(tenantDomain);
```

Comment:
Do we need to log anything related to this operation?

Reply:
I will add an audit log